### PR TITLE
[Import CSV] Authenticate calls to import CSV Cloud Function

### PR DESCRIPTION
**Pull request by gino-m**

### CODE REVIEW

**Comment by gino-m** [comment]
@parulraheja98 @gauravchalana1 @DaoyuT @DakshSaini @gyarill  This one up for grabs for review! Please reassign to yourself or ping this thread if you can review. Thanks!

**Comment by parulraheja98** [comment]
@gino-m , LGTM

**Comment by gino-m** [comment]
> @gino-m , LGTM

If it looks good please approve to unblock merge. Thanks!

**Review by parulraheja98 — APPROVED** [review]
SHIP IT 🚀

File: web-ng/src/app/services/auth/auth.service.ts
Comment (gino-m) [inline web-ng/src/app/services/auth/auth.service.ts:86]: Just tested on hosted Firebase and this is returning `error="invalid_token" error_description="The access token could not be verified"`. Trying to debug this but running out of leads. Please hold off on merging for now.

Comment (gino-m) [reply web-ng/src/app/services/auth/auth.service.ts:86]: Correction: the server is replying with this error in the "Authorization" response header.

Comment (gino-m) [reply web-ng/src/app/services/auth/auth.service.ts:86]: This also appears relevant:
https://firebase.google.com/docs/auth/admin/manage-cookies#verify_session_cookie_and_check_permissions

Comment (gino-m) [reply web-ng/src/app/services/auth/auth.service.ts:86]: And here:
https://firebase.google.com/docs/auth/web/auth-state-persistence#modifying_the_auth_state_persistence

**Comment by gino-m** [comment]
Update:
My understanding is that if we want export CSV to be able to stream downloads (#586), we'll need to access the Cloud Function via browser hyperlink. Since there's no way to set auth headers in hyperlinks, the only way we can authenticate would via cookies. Getting session cookies working ([instructions](https://firebase.google.com/docs/auth/admin/manage-cookies)) should get us there. If we start using session cookies for auth, this PR becomes obsolete, so closing it for now.

**Pull request closed without merging**
